src/lib/settings/settings.py

```diff
@@ -0,0 +1,280 @@
+            SettingFileNames.MONITORING_GROUPS: monitoring,
+            SettingFileNames.RECIPIENTS: recipients,
+        }
+        self._processed_settings = {
+            SettingFileNames.GENERAL: general,
+            SettingFileNames.MONITORING_GROUPS: monitoring,
+            SettingFileNames.RECIPIENTS: recipients,
+        }
+
+    @cached_property
+    def processed_settings(self):
+        # Add default metrics_extractor_role_arn
+        for m_env in self._processed_settings[SettingFileNames.GENERAL].get(
+            "monitored_environments", []
+        ):
+            if "metrics_extractor_role_arn" not in m_env:
+                m_env[
+                    "metrics_extractor_role_arn"
+                ] = self._get_default_metrics_extractor_role_arn(m_env["account_id"])
+
+        # Add default sla_seconds and minimum_number_of_runs
+        for m_env in self._processed_settings[SettingFileNames.MONITORING_GROUPS].get(
+            "monitoring_groups", []
+        ):
+            for m_res in SettingConfigs.RESOURCE_TYPES:
+                for res in m_env.get(m_res, []):
+                    if "sla_seconds" not in res:
+                        res["sla_seconds"] = 0
+                    if "minimum_number_of_runs" not in res:
+                        res["minimum_number_of_runs"] = 0
+
+        return self._processed_settings
+
+    @property
+    def general(self):
+        return self.processed_settings[SettingFileNames.GENERAL]
+
+    @cached_property
+    def monitoring_groups(self):
+        self._process_monitoring_groups()
+        return self.processed_settings[SettingFileNames.MONITORING_GROUPS]
+
+    @property
+    def recipients(self):
+        return self.processed_settings[SettingFileNames.RECIPIENTS]
+
+    # Processing methods
+    def _get_default_metrics_extractor_role_arn(self, account_id: str) -> str:
+        return f"arn:aws:iam::{account_id}:role/role-salmon-cross-account-extract-metrics-dev"
+
+    def _process_monitoring_groups(self):
+        # TODO: add wildcards replacement for all the resource types (glue, lambda, etc.)
+        pass
+
+    # Get raw settings by file name
+    def get_raw_settings(self, file_name: str) -> dict:
+        """Get raw settings by file name"""
+        return self._raw_settings[file_name]
+
+    # CDK methods
+    def get_monitored_account_ids(self) -> List[str]:
+        """Get monitored account_ids"""
+        return [
+            m_env["account_id"]
+            for m_env in self.processed_settings[SettingFileNames.GENERAL].get(
+                "monitored_environments", []
+            )
+        ]
+
+    def get_metrics_collection_interval_min(self) -> int:
+        """Get metrics_collection_interval_min"""
+        return self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"][
+            "metrics_collection_interval_min"
+        ]
+
+    # Lambda methods
+    def get_monitored_environment_name(self, account_id: str, region: str) -> List[str]:
```

Comment:
I guess this method shouldn't return list, but rather a single value.

account_id + region combination uniquely identifies monitored env

Reply:
done